joomla/joomla-cms
# [4.2] Fix check permission in mod_submenu

### Summary of Changes

The check of permissions for the menu items is different between `mod_submenu` (dashboards) and `mod_menu` (main menu).
administrator/modules/mod_submenu/src/Menu/Menu.php#L136 ff.
administrator/modules/mod_menu/src/Menu/CssMenu.php#L333 ff.

### Testing Instructions

Enable workflows in article options:
![image](https://user-images.githubusercontent.com/66922325/209428929-077dcf94-89e8-495b-a362-b744f97bbd35.png)

Check the content dashboard with a user with these permissions on com_content:
![image](https://user-images.githubusercontent.com/66922325/209428919-c06932b5-6222-48d4-907d-36d0fef4ce00.png)

### Actual result BEFORE applying this Pull Request

Categories is only displayed in the main menu (also applies to com_contact, ...)
![image](https://user-images.githubusercontent.com/66922325/209428940-bbc39803-a9fe-46f5-8ee8-ed785caa4b32.png)

When you click on workflows
![image](https://user-images.githubusercontent.com/66922325/209427004-f7f97d78-3e49-473d-b119-76e2d31123d5.png)

### Expected result AFTER applying this Pull Request

Categories is displayed in the module of the dashboard
Module containing workflows is not displayed (because no access to `core.manage.workflow` of com_content)
![image](https://user-images.githubusercontent.com/66922325/209428946-fe8caccf-d498-4cc3-a98e-c0c67729f4d2.png)

### Link to documentations
Please select:
